### PR TITLE
ENH: add average error to ErrorMap

### PR DESCRIPTION
**Pull request by lsqshr**
Add an average distance output to the `nipype.algorithms.metrics.ErrorMap`, which simpy get the square root average/the sum average of the errormap vector.

### CODE REVIEW

**Comment by satra** [comment]
@oesteban - could you please take a look at this? looks fine to me.

@lsqshr - could you please add an updated CHANGES file?

also it would be good add a test for this that uses nibabel to generate some fixed, known data and evaluates the metrics.

**Comment by lsqshr** [comment]
@satra Thanks for the comments. I will update the changes file and make some according tests.

**Comment by lsqshr** [comment]
@oesteban - in the following chunk of code in ErrorMap, were you just trying to calculate the euclidean distance voxel-wisely (`np.abs(diffvector)`)? Or it was just intended to calculate a norm of two images?
```
diffvector = (refvector-tstvector)
...
...
elif self.inputs.metric == 'euclidean':
            X = np.hstack((refvector, tstvector))
            errvector = np.linalg.norm(X, axis=1)
```
double check this because I am making some tests for it -- Thanks in advance

**Comment by oesteban** [comment]
@lsqshr you are right, but if I recall it well, I use ```np.linalg.norm``` for the case that ```refvector``` and ```tstvector``` are multispectral (meaning, they have more than one component per pixel).

The ```np.hstack``` is superfluous though, as you say, this can be simplified to ```errvector = np.linalg.norm(diffvector, axis=1)```

**Comment by satra** [comment]
perhaps we can also add the frobenius norm here and keep expanding the list of metrics.

**Comment by lsqshr** [comment]
- @oesteban  in this case I would change `np.hstack` to `diffvector` in this PR.
- @satra cool, I will add the frobenius norm in this PR for now

**Comment by lsqshr** [comment]
Just made the update and some tests with dummy 2*2*2 volumes

File: nipype/algorithms/metrics.py [hunk 1]
```diff
@@ -528,26 +530,30 @@ def _run_interface(self, runtime):
...
             msk = np.ones(shape=mapshape)
 
+        # Flatten both volumes and make the pixel differennce
         mskvector = msk.reshape(-1)
         msk_idxs = np.where(mskvector==1)
         refvector = ref_data.reshape(-1,comps)[msk_idxs].astype(np.float32)
         tstvector = tst_data.reshape(-1,comps)[msk_idxs].astype(np.float32)
         diffvector = (refvector-tstvector)
 
+        # Scale the diffrernce
```
Comment (oesteban) [inline nipype/algorithms/metrics.py]: Minimal typo (should read difference).

File: nipype/algorithms/metrics.py [hunk 2]
```diff
@@ -528,26 +530,30 @@ def _run_interface(self, runtime):
...
         if self.inputs.metric == 'sqeuclidean':
             errvector = diffvector**2
+            if (comps > 1):
+                errvector = np.sum(errvector, axis=1)
+            else:
+                errvector = np.squeeze(errvector)
         elif self.inputs.metric == 'euclidean':
-            X = np.hstack((refvector, tstvector))
-            errvector = np.linalg.norm(X, axis=1)
+            #X = np.hstack((refvector, tstvector))
```
Comment (oesteban) [inline nipype/algorithms/metrics.py]: You may safely remove this commented out line.

**Comment by oesteban** [comment]
@lsqshr the new code looks great, I've added some very minor comments. Additionally, we would need you to update your branch before I can merge the PR ([step 1](https://help.github.com/articles/configuring-a-remote-for-a-fork/), [step 2](https://help.github.com/articles/syncing-a-fork/)). If you need more help with that, do not hesitate to write me :).

**Comment by lsqshr** [comment]
@oesteban, thanks for reviewing the code and the comments. I have just made the according fixes and synced this branch with the upstream.

**Pull request merged**